openedx/core/lib/tests/test_xblock_utils.py

```diff
@@ -0,0 +1,208 @@
+    sanitize_html_id
+)
+
+
+@ddt.ddt
+class TestXblockUtils(SharedModuleStoreTestCase):
+    """
+    Tests for xblock utility functions.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestXblockUtils, cls).setUpClass()
+        cls.course_mongo = CourseFactory.create(
+            default_store=ModuleStoreEnum.Type.mongo,
+            org='TestX',
+            number='TS01',
+            run='2015'
+        )
+        cls.course_split = CourseFactory.create(
+            default_store=ModuleStoreEnum.Type.split,
+            org='TestX',
+            number='TS02',
+            run='2015'
+        )
+
+    def setUp(self):
+        super(TestXblockUtils, self).setUp()
+
+    def create_fragment(self, content=None):
+        """
+        Create a fragment.
+        """
+        fragment = Fragment(content)
+        fragment.add_css('body {background-color:red;}')
+        fragment.add_javascript('alert("Hi!");')
+        return fragment
+
+    def test_wrap_fragment(self):
+        """
+        Verify that wrap_fragment adds new content.
+        """
+        new_content = '<p>New Content<p>'
+        fragment = self.create_fragment()
+        wrapped_fragment = wrap_fragment(fragment, new_content)
+        self.assertEqual('<p>New Content<p>', wrapped_fragment.content)
+        self.assertEqual('body {background-color:red;}', wrapped_fragment.resources[0].data)
+        self.assertEqual('alert("Hi!");', wrapped_fragment.resources[1].data)
+
+    def test_request_token(self):
+        """
+        Verify that a proper token is returned.
+        """
+        request_with_token = RequestFactory().get('/')
+        request_with_token._xblock_token = '123'  # pylint: disable=protected-access
+        token = request_token(request_with_token)
+        self.assertEqual(token, '123')
+
+        request_without_token = RequestFactory().get('/')
+        token = request_token(request_without_token)
+        # Test to see if the token is an uuid1 hex value
+        test_uuid = uuid.UUID(token, version=1)
+        self.assertEqual(token, test_uuid.hex)
+
+    @ddt.data('course_mongo', 'course_split')
+    def test_wrap_xblock(self, course_id):
+        """
+        Verify that new content is added and the resources are the same.
+        """
+        fragment = self.create_fragment(u"<h1>Test!</h1>")
+        course = getattr(self, course_id)
+        test_wrap_output = wrap_xblock(
+            runtime_class='TestRuntime',
+            block=course,
+            view='baseview',
+            frag=fragment,
+            context=None,
+            usage_id_serializer=lambda usage_id: quote_slashes(unicode(usage_id)),
+            request_token=uuid.uuid1().get_hex()
+        )
+        self.assertIsInstance(test_wrap_output, Fragment)
+        self.assertIn('xblock-baseview', test_wrap_output.content)
+        self.assertIn('data-runtime-class="TestRuntime"', test_wrap_output.content)
+        if course_id == 'course_mongo':
```

Comment:
the better pattern for this is to include the `data-usage-id` and then use `ddt.unpack`: http://ddt.readthedocs.org/en/latest/example.html.

So it would look like:

```python
@ddt.data(
    ('course_mongo',  'data-usage-id="i4x:;_;_TestX;_TS01;_course;_2015"'),
    ('course_split', 'data-usage-id="block-v1:TestX+TS02+2015+type@course+block@course"')
)
@ddt.unpack
def test_wrap_xblock(self, course_id, data-usage-id):
    ...
```

I'm happy to add this as a commit myself, if you'd like :)